tests/queries/0_stateless/01925_join_materialized_columns.sql

```diff
@@ -51,5 +52,6 @@ SELECT t1.time as talias FROM t1 JOIN t2 ON talias = t2.time;
 SELECT t1.time as talias FROM t1 JOIN t2 ON talias = t2.time_alias;
 SELECT t2.time as talias FROM t1 JOIN t2 ON t1.time = talias;
 SELECT t2.time as talias FROM t1 JOIN t2 ON t1.time_alias = talias;
-SELECT time as talias FROM t1 JOIN t2 ON t1.time = talias; -- { serverError AMBIGUOUS_COLUMN_NAME }
-SELECT time as talias FROM t1 JOIN t2 ON talias = t2.time; -- { serverError AMBIGUOUS_COLUMN_NAME }
+SELECT time as talias FROM t1 JOIN t2 ON t1.time = talias; -- { serverError AMBIGUOUS_COLUMN_NAME, INVALID_JOIN_ON_EXPRESSION }
+SELECT time as talias FROM t1 JOIN t2 ON talias = t2.time settings allow_experimental_analyzer=0; -- { serverError AMBIGUOUS_COLUMN_NAME }
```

Comment:
I checked it once more and now I see that the error is about `time` column, but not identifier `talias`. We resolve alias defined in SELECT and used in ON section, so should also support it in analyzer (same for using https://github.com/ClickHouse/ClickHouse/pull/60182 will work only with the setting, though).

But as for the `time` identifier we resolve it from left table due to `single_join_prefer_left_table` is set by default.